tests/test_sanity.py

```diff
@@ -144,14 +149,60 @@ def test_from_container(
     runtime_specific_args: List[str],
     gprofiler_docker_image: Image,
     output_directory: Path,
+    output_collapsed: Path,
     assert_collapsed: AssertInCollapsed,
     assert_app_id: Callable,
     profiler_flags: List[str],
 ) -> None:
     _ = application_pid  # Fixture only used for running the application.
     _ = assert_app_id  # Required for mypy unused argument warning
     collapsed_text = run_gprofiler_in_container_for_one_session(
-        docker_client, gprofiler_docker_image, output_directory, runtime_specific_args, profiler_flags
+        docker_client, gprofiler_docker_image, output_directory, output_collapsed, runtime_specific_args, profiler_flags
     )
     collapsed = parse_one_collapsed(collapsed_text)
     assert_collapsed(collapsed)
+
+
+@pytest.mark.parametrize(
+    "runtime,profiler_type",
+    [
+        ("java", "ap"),
+        ("python", "py-spy"),
+        ("ruby", "rbspy"),
+    ],
+)
+def test_from_container_spawned_process(
+    docker_client: DockerClient,
+    runtime_specific_args: List[str],
+    gprofiler_docker_image: Image,
+    output_directory: Path,
+    output_collapsed: Path,
+    assert_collapsed: AssertInCollapsed,
+    profiler_flags: List[str],
+    application_factory: Callable[[], _GeneratorContextManager],
+) -> None:
+    args = runtime_specific_args.copy()
+    if "-d" in args:
+        args[args.index("-d") + 1] = "10"
+    else:
+        args.extend(["-d", "10"])
```

Comment:
```suggestion
    args.extend(["-d", "10"])
```

it's okay to supply it multiple times in any case